// @aws-sdk/client-iot-data-plane/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "iotdata",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class IoTDataPlaneClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultIoTDataPlaneHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class IoTDataPlaneServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, IoTDataPlaneServiceException.prototype);
    }
}

class ConflictException extends IoTDataPlaneServiceException {
    name = "ConflictException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
    }
}
class ForbiddenException extends IoTDataPlaneServiceException {
    name = "ForbiddenException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ForbiddenException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ForbiddenException.prototype);
    }
}
class InternalFailureException extends IoTDataPlaneServiceException {
    name = "InternalFailureException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "InternalFailureException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalFailureException.prototype);
    }
}
class InvalidRequestException extends IoTDataPlaneServiceException {
    name = "InvalidRequestException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidRequestException.prototype);
    }
}
class ResourceNotFoundException extends IoTDataPlaneServiceException {
    name = "ResourceNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
    }
}
class ThrottlingException extends IoTDataPlaneServiceException {
    name = "ThrottlingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
    }
}
class MethodNotAllowedException extends IoTDataPlaneServiceException {
    name = "MethodNotAllowedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "MethodNotAllowedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MethodNotAllowedException.prototype);
    }
}
class ServiceUnavailableException extends IoTDataPlaneServiceException {
    name = "ServiceUnavailableException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "ServiceUnavailableException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceUnavailableException.prototype);
    }
}
class UnauthorizedException extends IoTDataPlaneServiceException {
    name = "UnauthorizedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "UnauthorizedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnauthorizedException.prototype);
    }
}
class UnsupportedDocumentEncodingException extends IoTDataPlaneServiceException {
    name = "UnsupportedDocumentEncodingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "UnsupportedDocumentEncodingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnsupportedDocumentEncodingException.prototype);
    }
}
class RequestEntityTooLargeException extends IoTDataPlaneServiceException {
    name = "RequestEntityTooLargeException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "RequestEntityTooLargeException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, RequestEntityTooLargeException.prototype);
    }
}

const _CE = "ConflictException";
const _DC = "DeleteConnection";
const _DCR = "DeleteConnectionRequest";
const _DTS = "DeleteThingShadow";
const _DTSR = "DeleteThingShadowRequest";
const _DTSRe = "DeleteThingShadowResponse";
const _FE = "ForbiddenException";
const _GRM = "GetRetainedMessage";
const _GRMR = "GetRetainedMessageRequest";
const _GRMRe = "GetRetainedMessageResponse";
const _GTS = "GetThingShadow";
const _GTSR = "GetThingShadowRequest";
const _GTSRe = "GetThingShadowResponse";
const _IFE = "InternalFailureException";
const _IRE = "InvalidRequestException";
const _LNSFT = "ListNamedShadowsForThing";
const _LNSFTR = "ListNamedShadowsForThingRequest";
const _LNSFTRi = "ListNamedShadowsForThingResponse";
const _LRM = "ListRetainedMessages";
const _LRMR = "ListRetainedMessagesRequest";
const _LRMRi = "ListRetainedMessagesResponse";
const _MNAE = "MethodNotAllowedException";
const _P = "Publish";
const _PR = "PublishRequest";
const _RETLE = "RequestEntityTooLargeException";
const _RML = "RetainedMessageList";
const _RMS = "RetainedMessageSummary";
const _RNFE = "ResourceNotFoundException";
const _SJUP = "SynthesizedJsonUserProperties";
const _SUE = "ServiceUnavailableException";
const _TE = "ThrottlingException";
const _UDEE = "UnsupportedDocumentEncodingException";
const _UE = "UnauthorizedException";
const _UTS = "UpdateThingShadow";
const _UTSR = "UpdateThingShadowRequest";
const _UTSRp = "UpdateThingShadowResponse";
const _a = "application/json";
const _c = "client";
const _cD = "correlationData";
const _cI = "clientId";
const _cS = "cleanSession";
const _cT = "contentType";
const _e = "error";
const _h = "http";
const _hE = "httpError";
const _hH = "httpHeader";
const _hQ = "httpQuery";
const _lMT = "lastModifiedTime";
const _m = "message";
const _mE = "messageExpiry";
const _mR = "maxResults";
const _mT = "mediaType";
const _n = "name";
const _nT = "nextToken";
const _p = "payload";
const _pFI = "payloadFormatIndicator";
const _pS = "pageSize";
const _pSa = "payloadSize";
const _pWM = "preventWillMessage";
const _q = "qos";
const _r = "results";
const _rT = "retainedTopics";
const _rTe = "responseTopic";
const _re = "retain";
const _s = "server";
const _sN = "shadowName";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.iotdataplane";
const _t = "topic";
const _tN = "thingName";
const _ti = "timestamp";
const _uP = "userProperties";
const _xamcd = "x-amz-mqtt5-correlation-data";
const _xampfi = "x-amz-mqtt5-payload-format-indicator";
const _xamup = "x-amz-mqtt5-user-properties";
const n0 = "com.amazonaws.iotdataplane";
var SynthesizedJsonUserProperties = [0, n0, _SJUP, { [_mT]: _a }, 0];
var ConflictException$ = [-3, n0, _CE,
    { [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ConflictException$, ConflictException);
var DeleteConnectionRequest$ = [3, n0, _DCR,
    0,
    [_cI, _cS, _pWM],
    [[0, 1], [2, { [_hQ]: _cS }], [2, { [_hQ]: _pWM }]], 1
];
var DeleteThingShadowRequest$ = [3, n0, _DTSR,
    0,
    [_tN, _sN],
    [[0, 1], [0, { [_hQ]: _n }]], 1
];
var DeleteThingShadowResponse$ = [3, n0, _DTSRe,
    0,
    [_p],
    [[21, 16]], 1
];
var ForbiddenException$ = [-3, n0, _FE,
    { [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ForbiddenException$, ForbiddenException);
var GetRetainedMessageRequest$ = [3, n0, _GRMR,
    0,
    [_t],
    [[0, 1]], 1
];
var GetRetainedMessageResponse$ = [3, n0, _GRMRe,
    0,
    [_t, _p, _q, _lMT, _uP],
    [0, 21, 1, 1, 21]
];
var GetThingShadowRequest$ = [3, n0, _GTSR,
    0,
    [_tN, _sN],
    [[0, 1], [0, { [_hQ]: _n }]], 1
];
var GetThingShadowResponse$ = [3, n0, _GTSRe,
    0,
    [_p],
    [[21, 16]]
];
var InternalFailureException$ = [-3, n0, _IFE,
    { [_e]: _s, [_hE]: 500 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InternalFailureException$, InternalFailureException);
var InvalidRequestException$ = [-3, n0, _IRE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InvalidRequestException$, InvalidRequestException);
var ListNamedShadowsForThingRequest$ = [3, n0, _LNSFTR,
    0,
    [_tN, _nT, _pS],
    [[0, 1], [0, { [_hQ]: _nT }], [1, { [_hQ]: _pS }]], 1
];
var ListNamedShadowsForThingResponse$ = [3, n0, _LNSFTRi,
    0,
    [_r, _nT, _ti],
    [64 | 0, 0, 1]
];
var ListRetainedMessagesRequest$ = [3, n0, _LRMR,
    0,
    [_nT, _mR],
    [[0, { [_hQ]: _nT }], [1, { [_hQ]: _mR }]]
];
var ListRetainedMessagesResponse$ = [3, n0, _LRMRi,
    0,
    [_rT, _nT],
    [() => RetainedMessageList, 0]
];
var MethodNotAllowedException$ = [-3, n0, _MNAE,
    { [_e]: _c, [_hE]: 405 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(MethodNotAllowedException$, MethodNotAllowedException);
var PublishRequest$ = [3, n0, _PR,
    0,
    [_t, _q, _re, _p, _uP, _pFI, _cT, _rTe, _cD, _mE],
    [[0, 1], [1, { [_hQ]: _q }], [2, { [_hQ]: _re }], [21, 16], [() => SynthesizedJsonUserProperties, { [_hH]: _xamup }], [0, { [_hH]: _xampfi }], [0, { [_hQ]: _cT }], [0, { [_hQ]: _rTe }], [0, { [_hH]: _xamcd }], [1, { [_hQ]: _mE }]], 1
];
var RequestEntityTooLargeException$ = [-3, n0, _RETLE,
    { [_e]: _c, [_hE]: 413 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(RequestEntityTooLargeException$, RequestEntityTooLargeException);
var ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ResourceNotFoundException$, ResourceNotFoundException);
var RetainedMessageSummary$ = [3, n0, _RMS,
    0,
    [_t, _pSa, _q, _lMT],
    [0, 1, 1, 1]
];
var ServiceUnavailableException$ = [-3, n0, _SUE,
    { [_e]: _s, [_hE]: 503 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ServiceUnavailableException$, ServiceUnavailableException);
var ThrottlingException$ = [-3, n0, _TE,
    { [_e]: _c, [_hE]: 429 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ThrottlingException$, ThrottlingException);
var UnauthorizedException$ = [-3, n0, _UE,
    { [_e]: _c, [_hE]: 401 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(UnauthorizedException$, UnauthorizedException);
var UnsupportedDocumentEncodingException$ = [-3, n0, _UDEE,
    { [_e]: _c, [_hE]: 415 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(UnsupportedDocumentEncodingException$, UnsupportedDocumentEncodingException);
var UpdateThingShadowRequest$ = [3, n0, _UTSR,
    0,
    [_tN, _p, _sN],
    [[0, 1], [21, 16], [0, { [_hQ]: _n }]], 2
];
var UpdateThingShadowResponse$ = [3, n0, _UTSRp,
    0,
    [_p],
    [[21, 16]]
];
var __Unit = "unit";
var IoTDataPlaneServiceException$ = [-3, _sm, "IoTDataPlaneServiceException", 0, [], []];
schema.TypeRegistry.for(_sm).registerError(IoTDataPlaneServiceException$, IoTDataPlaneServiceException);
var RetainedMessageList = [1, n0, _RML,
    0, () => RetainedMessageSummary$
];
var DeleteConnection$ = [9, n0, _DC,
    { [_h]: ["DELETE", "/connections/{clientId}", 200] }, () => DeleteConnectionRequest$, () => __Unit
];
var DeleteThingShadow$ = [9, n0, _DTS,
    { [_h]: ["DELETE", "/things/{thingName}/shadow", 200] }, () => DeleteThingShadowRequest$, () => DeleteThingShadowResponse$
];
var GetRetainedMessage$ = [9, n0, _GRM,
    { [_h]: ["GET", "/retainedMessage/{topic}", 200] }, () => GetRetainedMessageRequest$, () => GetRetainedMessageResponse$
];
var GetThingShadow$ = [9, n0, _GTS,
    { [_h]: ["GET", "/things/{thingName}/shadow", 200] }, () => GetThingShadowRequest$, () => GetThingShadowResponse$
];
var ListNamedShadowsForThing$ = [9, n0, _LNSFT,
    { [_h]: ["GET", "/api/things/shadow/ListNamedShadowsForThing/{thingName}", 200] }, () => ListNamedShadowsForThingRequest$, () => ListNamedShadowsForThingResponse$
];
var ListRetainedMessages$ = [9, n0, _LRM,
    { [_h]: ["GET", "/retainedMessage", 200] }, () => ListRetainedMessagesRequest$, () => ListRetainedMessagesResponse$
];
var Publish$ = [9, n0, _P,
    { [_h]: ["POST", "/topics/{topic}", 200] }, () => PublishRequest$, () => __Unit
];
var UpdateThingShadow$ = [9, n0, _UTS,
    { [_h]: ["POST", "/things/{thingName}/shadow", 200] }, () => UpdateThingShadowRequest$, () => UpdateThingShadowResponse$
];

class DeleteConnectionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("IotMoonrakerService", "DeleteConnection", {})
    .n("IoTDataPlaneClient", "DeleteConnectionCommand")
    .sc(DeleteConnection$)
    .build() {
}

class DeleteThingShadowCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("IotMoonrakerService", "DeleteThingShadow", {})
    .n("IoTDataPlaneClient", "DeleteThingShadowCommand")
    .sc(DeleteThingShadow$)
    .build() {
}

class GetRetainedMessageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("IotMoonrakerService", "GetRetainedMessage", {})
    .n("IoTDataPlaneClient", "GetRetainedMessageCommand")
    .sc(GetRetainedMessage$)
    .build() {
}

class GetThingShadowCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("IotMoonrakerService", "GetThingShadow", {})
    .n("IoTDataPlaneClient", "GetThingShadowCommand")
    .sc(GetThingShadow$)
    .build() {
}

class ListNamedShadowsForThingCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("IotMoonrakerService", "ListNamedShadowsForThing", {})
    .n("IoTDataPlaneClient", "ListNamedShadowsForThingCommand")
    .sc(ListNamedShadowsForThing$)
    .build() {
}

class ListRetainedMessagesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("IotMoonrakerService", "ListRetainedMessages", {})
    .n("IoTDataPlaneClient", "ListRetainedMessagesCommand")
    .sc(ListRetainedMessages$)
    .build() {
}

class PublishCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("IotMoonrakerService", "Publish", {})
    .n("IoTDataPlaneClient", "PublishCommand")
    .sc(Publish$)
    .build() {
}

class UpdateThingShadowCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("IotMoonrakerService", "UpdateThingShadow", {})
    .n("IoTDataPlaneClient", "UpdateThingShadowCommand")
    .sc(UpdateThingShadow$)
    .build() {
}

const paginateListRetainedMessages = core.createPaginator(IoTDataPlaneClient, ListRetainedMessagesCommand, "nextToken", "nextToken", "maxResults");

const commands = {
    DeleteConnectionCommand,
    DeleteThingShadowCommand,
    GetRetainedMessageCommand,
    GetThingShadowCommand,
    ListNamedShadowsForThingCommand,
    ListRetainedMessagesCommand,
    PublishCommand,
    UpdateThingShadowCommand,
};
const paginators = {
    paginateListRetainedMessages,
};
class IoTDataPlane extends IoTDataPlaneClient {
}
smithyClient.createAggregatedClient(commands, IoTDataPlane, { paginators });

const PayloadFormatIndicator = {
    UNSPECIFIED_BYTES: "UNSPECIFIED_BYTES",
    UTF8_DATA: "UTF8_DATA",
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.ConflictException = ConflictException;
exports.ConflictException$ = ConflictException$;
exports.DeleteConnection$ = DeleteConnection$;
exports.DeleteConnectionCommand = DeleteConnectionCommand;
exports.DeleteConnectionRequest$ = DeleteConnectionRequest$;
exports.DeleteThingShadow$ = DeleteThingShadow$;
exports.DeleteThingShadowCommand = DeleteThingShadowCommand;
exports.DeleteThingShadowRequest$ = DeleteThingShadowRequest$;
exports.DeleteThingShadowResponse$ = DeleteThingShadowResponse$;
exports.ForbiddenException = ForbiddenException;
exports.ForbiddenException$ = ForbiddenException$;
exports.GetRetainedMessage$ = GetRetainedMessage$;
exports.GetRetainedMessageCommand = GetRetainedMessageCommand;
exports.GetRetainedMessageRequest$ = GetRetainedMessageRequest$;
exports.GetRetainedMessageResponse$ = GetRetainedMessageResponse$;
exports.GetThingShadow$ = GetThingShadow$;
exports.GetThingShadowCommand = GetThingShadowCommand;
exports.GetThingShadowRequest$ = GetThingShadowRequest$;
exports.GetThingShadowResponse$ = GetThingShadowResponse$;
exports.InternalFailureException = InternalFailureException;
exports.InternalFailureException$ = InternalFailureException$;
exports.InvalidRequestException = InvalidRequestException;
exports.InvalidRequestException$ = InvalidRequestException$;
exports.IoTDataPlane = IoTDataPlane;
exports.IoTDataPlaneClient = IoTDataPlaneClient;
exports.IoTDataPlaneServiceException = IoTDataPlaneServiceException;
exports.IoTDataPlaneServiceException$ = IoTDataPlaneServiceException$;
exports.ListNamedShadowsForThing$ = ListNamedShadowsForThing$;
exports.ListNamedShadowsForThingCommand = ListNamedShadowsForThingCommand;
exports.ListNamedShadowsForThingRequest$ = ListNamedShadowsForThingRequest$;
exports.ListNamedShadowsForThingResponse$ = ListNamedShadowsForThingResponse$;
exports.ListRetainedMessages$ = ListRetainedMessages$;
exports.ListRetainedMessagesCommand = ListRetainedMessagesCommand;
exports.ListRetainedMessagesRequest$ = ListRetainedMessagesRequest$;
exports.ListRetainedMessagesResponse$ = ListRetainedMessagesResponse$;
exports.MethodNotAllowedException = MethodNotAllowedException;
exports.MethodNotAllowedException$ = MethodNotAllowedException$;
exports.PayloadFormatIndicator = PayloadFormatIndicator;
exports.Publish$ = Publish$;
exports.PublishCommand = PublishCommand;
exports.PublishRequest$ = PublishRequest$;
exports.RequestEntityTooLargeException = RequestEntityTooLargeException;
exports.RequestEntityTooLargeException$ = RequestEntityTooLargeException$;
exports.ResourceNotFoundException = ResourceNotFoundException;
exports.ResourceNotFoundException$ = ResourceNotFoundException$;
exports.RetainedMessageSummary$ = RetainedMessageSummary$;
exports.ServiceUnavailableException = ServiceUnavailableException;
exports.ServiceUnavailableException$ = ServiceUnavailableException$;
exports.ThrottlingException = ThrottlingException;
exports.ThrottlingException$ = ThrottlingException$;
exports.UnauthorizedException = UnauthorizedException;
exports.UnauthorizedException$ = UnauthorizedException$;
exports.UnsupportedDocumentEncodingException = UnsupportedDocumentEncodingException;
exports.UnsupportedDocumentEncodingException$ = UnsupportedDocumentEncodingException$;
exports.UpdateThingShadow$ = UpdateThingShadow$;
exports.UpdateThingShadowCommand = UpdateThingShadowCommand;
exports.UpdateThingShadowRequest$ = UpdateThingShadowRequest$;
exports.UpdateThingShadowResponse$ = UpdateThingShadowResponse$;
exports.paginateListRetainedMessages = paginateListRetainedMessages;
